doc/dev/installation.rst

```diff
@@ -70,6 +70,79 @@ Building from source
 To build Catalyst from source, developers should follow the instructions provided below for building
 all three modules: the Python frontend, the MLIR compiler, and the runtime library.
 
+
+Just Tell Me How To Build Catalyst From Source
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+(This section title is a tribute to the llvm new pass manager guide's section `Just Tell Me How To Run The Default Optimization Pipeline With The New Pass Manager <https://llvm.org/docs/NewPassManager.html#just-tell-me-how-to-run-the-default-optimization-pipeline-with-the-new-pass-manager>`_)
+
+On **Linux Debian/Ubuntu**:
+
+.. code-block:: console
+
+  # Install common requirements
+  sudo apt install clang lld ccache libomp-dev ninja-build make cmake 
+
+  # Clone the Catalyst repository  
+  git clone --recurse-submodules --shallow-submodules https://github.com/PennyLaneAI/catalyst.git
+
+  # Install specific requirements for Catalyst
+  pip install -r requirements.txt
+
+  # Build Catalyst
+  make all
+
+  # If you are building Catalyst components in custom locations, set environment variables. 
+  # This step should not be required if you are simply following the instructions above. 
+  export PYTHONPATH="$PWD/mlir/build/python_packages/quantum:$PYTHONPATH"
+  export RUNTIME_LIB_DIR="$PWD/runtime/build/lib"
+  export MLIR_LIB_DIR="$PWD/mlir/llvm-project/build/lib"
+  export ENZYME_LIB_DIR="$PWD/mlir/Enzyme/build/Enzyme"
+  export PATH="$PWD/mlir/llvm-project/build/bin:$PWD/mlir/mlir-hlo/mhlo-build/bin:$PWD/mlir/build/bin:$PATH"
+
+  # Optional: test that everything is built properly
+  make test
+
+  # Optional: build and test documentation for Catalyst
+  pip install -r doc/requirements.txt
+  sudo apt install doxygen pandoc
+
+
+On **macOS**:
```

Comment:
@paul0403 It would be great to create a tab object for different platforms that would allow you to select the instructions you want (in the entire document) without having to scroll / without displaying the other options. @josh146 might be able to help figuring out how this can be done with sphinx?

Reply:
Although this is looking pretty promising already: https://sphinx-tabs.readthedocs.io/en/latest/#group-tabs

Reply:
💯

Reply:
@dime10 yep Sphinx-Tabs was on my backlog to update the documentation, so that the reader can pick and see only the commands relevant to them.